app_dart/lib/src/request_handlers/presubmit_luci_subscription_v2.dart

```diff
@@ -0,0 +1,168 @@
+import 'package:cocoon_service/src/request_handling/subscription_handler_v2.dart';
+import 'package:cocoon_service/src/service/github_checks_service_v2.dart';
+import 'package:cocoon_service/src/service/luci_build_service_v2.dart';
+import 'package:cocoon_service/src/service/scheduler_v2.dart';
+import 'package:github/github.dart';
+import 'package:meta/meta.dart';
+
+import '../model/appengine/commit.dart';
+import '../model/ci_yaml/ci_yaml.dart';
+import '../model/ci_yaml/target.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+import '../service/config.dart';
+import '../service/logging.dart';
+
+/// An endpoint for listening to LUCI status updates for scheduled builds.
+///
+/// [ScheduleBuildRequest.notify] property is set to tell LUCI to use this
+/// PubSub topic. LUCI then publishes updates about build status to that topic,
+/// which we listen to on the github-updater subscription. When new messages
+/// arrive, they are posted to this web service.
+///
+/// The PubSub subscription is set up here:
+/// https://console.cloud.google.com/cloudpubsub/subscription/detail/build-bucket-presubmit-sub?project=flutter-dashboard
+///
+/// This endpoint is responsible for updating GitHub with the status of
+/// completed builds from LUCI.
+@immutable
+class PresubmitLuciSubscriptionV2 extends SubscriptionHandlerV2 {
+  /// Creates an endpoint for listening to LUCI status updates.
+  const PresubmitLuciSubscriptionV2({
+    required super.cache,
+    required super.config,
+    required this.scheduler,
+    required this.luciBuildService,
+    required this.githubChecksService,
+    AuthenticationProvider? authProvider,
+  }) : super(subscriptionName: 'build-bucket-presubmit-sub');
+
+  final LuciBuildServiceV2 luciBuildService;
+  final GithubChecksServiceV2 githubChecksService;
+  final SchedulerV2 scheduler;
+
+  @override
+  Future<Body> post() async {
+    if (message.data == null) {
+      log.info('no data in message');
+      return Body.empty;
+    }
+
+    final bbv2.PubSubCallBack pubSubCallBack = bbv2.PubSubCallBack();
+    pubSubCallBack.mergeFromProto3Json(jsonDecode(message.data!) as Map<String, dynamic>);
+
+    final bbv2.BuildsV2PubSub buildsV2PubSub = pubSubCallBack.buildPubsub;
+
+    if (!buildsV2PubSub.hasBuild()) {
+      log.info('no build information in message');
+      return Body.empty;
+    }
+
+    final bbv2.Build build = buildsV2PubSub.build;
+
+    final String builderName = build.builder.builder;
+
+    final List<bbv2.StringPair> tags = build.tags;
+
+    log.fine('Available tags: ${tags.toString()}');
+
+    // Skip status update if we can not get the sha tag.
+    if (tags.where((element) => element.key == 'buildset').isEmpty) {
+      log.warning('Buildset tag not included, skipping Status Updates');
+      return Body.empty;
+    }
+
+    log.fine('Setting status (${build.status.toString()}) for $builderName');
+
+    if (pubSubCallBack.hasUserData()) {
```

Comment:
```suggestion
    if (!pubSubCallBack.hasUserData()) {
      return Body.empty;
    }
```

https://dcm.dev/docs/rules/common/avoid-nested-conditional-expressions/

Reply:
Refactored to remove 1 level of cyclomatic complexity.